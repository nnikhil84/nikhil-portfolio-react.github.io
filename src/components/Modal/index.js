import React from "react";
// import { FaGithubSquare, FaChrome } from "react-icons/fa";
// import { IconContext } from "react-icons";

function Modal({ currentPhoto, onClose }) {
  const { name, category, link, description, index } = currentPhoto;

  return (
    <div className="modalBackdrop">
      <div className="modalContainer">
        <h3 className="modalTitle">{name}</h3>
        <button type="button" onClick={onClose}>
          Click to close this Page
        </button>

        {/* <IconContext.Provider value={{ className: "icon-color", size: 20 }}> */}
          <div className="wrapper">
            <a href="https://github.com/nnikhil84">
              {" "}
              Repository for Project
              {/* <FaGithubSquare /> */}
            </a>

            <a href={link}>
              {" "}
              Application Deployed at
              {/* <FaChrome /> */}
            </a>
          </div>
        {/* </IconContext.Provider> */}
        <p>{description}</p>
        <img
          alt="current category"
          src={require(`../../assets/large/${category}/${index}.jpg`)}
        />
      </div>
    </div>
  );
}

export default Modal;
