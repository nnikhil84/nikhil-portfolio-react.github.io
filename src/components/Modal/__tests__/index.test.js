import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup, fireEvent } from "@testing-library/react";
import Modal from "..";

const mockCurrentPhoto = {
  name: "Grocery aisle",
  description:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie",
  category: "commercial",
  index: 0,
};
const mockToggleModal = jest.fn();

afterEach(cleanup);

describe("Modal component", () => {
  it("renders", () => {
    render(<Modal currentPhoto={mockCurrentPhoto} onClose={mockToggleModal} />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Modal currentPhoto={mockCurrentPhoto} onClose={mockToggleModal} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Click Event", () => {
  it("calls onClose handler", () => {
    const { getByText } = render(
      <Modal currentPhoto={mockCurrentPhoto} onClose={mockToggleModal} />
    );

    fireEvent.click(getByText("Close this modal"));

    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
});
