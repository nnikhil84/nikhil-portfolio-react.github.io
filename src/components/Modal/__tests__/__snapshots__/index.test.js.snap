// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Grocery aisle
      </h3>
      <img
        alt="current category"
        src="0.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
